src/services/CertificateGenerationService.ts

```diff
@@ -507,9 +541,49 @@ class CertificateGenerationService {
                 };
                 console.log("CHECK HERE DOCGENPAYLOAD -> ", docGenPayloadAdr);
                 return docGenPayloadAdr;
+            case CERTIFICATE_DATA.IVA_DATA:
+                const ivaFailDetailsForDocGen = {
```

Comment:
1. should this vehicleTrailerNrNo say vehicleTrailerOrNo?

2. it's probably not a good idea to do data processing while constructing an object, I'd suggest having it in a variables above the object and may help with debugging if an issue down the line comes up. For example:
const vehicle = trailer ? trailer id : vrm.
3. A similar thing to the above, could be said for the reapplication date and date - makes it easier to read. also, depending on if you make the reapplication date the same, yo ucoud use the object literal since the name may be the same
4. You could also inline the returned object